### PR TITLE
[ci] Add a web version of Dart unit tests

### PR DESCRIPTION
**Pull request by stuartmorgan**
Adds new LUCI targets in bringup mode to run all possible Dart unit tests in Chrome.

This is a new test (see linked issue), not a port of a Cirrus test, so it involves changes to tooling and packages:
- The tooling now accepts an explicit platform. The default behavior if none is provided is the previous behavior of running in VM for everything but web plugin implementations (since that's convenient locally).
- The tooling now has a basic understanding of `dart_test.yaml` `test_on` directives, to know when to skip.
- Packages that don't support web have opt-out files.
- Packages that do support web but have a few tests that fail on web have those tests opted out.
- Packages that do support web but have a lot of failures on web have temporary opt-out files with TODOs + issue links.

Most of https://github.com/flutter/flutter/issues/128979

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -93,48 +93,32 @@ targets:
       version_file: flutter_master.version
 
   - name: Linux dart_unit_test_shard_1 master
```
Comment (keyonghan) [inline .ci.yaml:95]: Do you want to change the name to reflect `web`: `Linux web_dart_unit_test_shard_1 master`

File: .ci.yaml [hunk 2]
```diff
@@ -93,48 +93,32 @@ targets:
...
-    bringup: true # New target
```
Comment (keyonghan) [inline .ci.yaml:96]: As this seems a new target, shall we keep the `bringup`?

File: .ci.yaml (hunk 1 above)
Comment (stuartmorgan) [reply .ci.yaml:95]: This was me hacking the web version into the existing config so I could run in CI without mucking around with `led`. The non-draft version once I have things passing will be new, bringup connfigs.

**Comment by stuartmorgan** [comment]
This is now ready for review. (Those of you who are incidentally pulled in because of minor changes to the package test configs are welcome to ignore this.)

**Review by chunhtai — APPROVED** [review]
go_router_builder lgtm

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/test/types/bitmap_test.dart [hunk 3]
```diff
@@ -185,7 +185,9 @@ void main() {
 
           expect((mip.toJson() as List<dynamic>)[2], 1);
           expect((scaled.toJson() as List<dynamic>)[2], 3);
-        });
+        },
+            // TODO(stuartmorgan): Investigate timeout on web.
```
Comment (tarrinneal) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/test/types/bitmap_test.dart:189]: Does this need an issue?

**Review by tarrinneal — APPROVED** [review]

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/test/types/bitmap_test.dart (hunk 3 above)
Comment (stuartmorgan) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/test/types/bitmap_test.dart:189]: Based on a comment from David later I think we just need to change this to inline image data. I'll do a fast-follow on this.

**Pull request merged**